### PR TITLE
fix(CLAPI): APPLYCFG Sync Task for Remote Servers

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR fix an issue where CLAPI APPLYCFG on a Poller doesnt sync task on a parent remote server.

**Fixes** # MON-5561

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 1]
```diff
@@ -543,10 +545,66 @@ public function cfgMove($variables)
...
+                    'pollers' => []
+                ];
+
+                if (!empty($linkedResults)) {
+                    $exportParams['pollers'] = array_column($linkedResults, 'id');
+                } else {
+                    $exportParams['pollers'] = [$remote['id']];
+                }
+
+                $dependencyInjector['centreon.taskservice']->addTask(Task::TYPE_EXPORT, ['params' => $exportParams]);
```
Comment (kduret) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:604]: use constant for `centreon.taskservice`  : https://github.com/centreon/centreon/blob/master/src/CentreonRemote/ServiceProvider.php#L44

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 2]
```diff
@@ -425,12 +429,12 @@ public function pollerGenerate($variables, $login, $password)
      * Move configuration files to servers
      * @param unknown_type $variables
      */
-    public function cfgMove($variables)
+    public function cfgMove($variables = null)
     {
-        global $pearDB, $pearDBO;
+        global $pearDB, $pearDBO, $dependencyInjector;
```
Comment (callapa) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:434]: Don't use the DI like that. Inject the necessary class instance directly into the method.

**Review by callapa — CHANGES_REQUESTED** [review]

File: www/class/centreon-clapi/centreon.Config.Poller.class.php (hunk 2 above)
Comment (jeremyjaouen) [reply www/class/centreon-clapi/centreon.Config.Poller.class.php]: I found a better way than we discuss yesterday. I haven't noticed we called the DI into the constructor of the class so I've directly use as $this->dependencyInjector

**Review by kduret — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Review by callapa — APPROVED** [review]

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 3]
```diff
@@ -543,10 +549,69 @@ public function cfgMove($variables)
...
-            exec("echo 'SENDCFGFILE:" . $host['id'] . "' >> " . $this->centcore_pipe, $stdout, $return);
-            if (!isset($msg_copy)) {
-                $msg_copy = "";
+            /**
+             * Get Parent Remote Servers of the Poller
+             */
+            $statementRemotes = $pearDB->prepare(
+                'SELECT ns.id
+                FROM nagios_server AS ns
+                JOIN platform_topology AS pt ON (ns.id = pt.server_id)
```
Comment (yannbeulque) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:558]: platform_topology don't existe on centreon-web 20.04.xx, add in 20.10.xx https://github.com/centreon/centreon/blob/98185b835f72e0f414999c3f476d8cb8601bd185/www/install/sql/centreon/Update-DB-20.10.0-beta.1.sql#L28

Clapi over http return status code 500 : 
image

Comment (kduret) [reply www/class/centreon-clapi/centreon.Config.Poller.class.php:558]: this is fixed here : https://github.com/centreon/centreon/commit/fbe2f11e02c76556e6f2605be61daffc566d5220
And will be release with 20.04.11 :)

**Pull request merged**